### DISCUSSION TITLE
RTE validations not working

### DISCUSSION THREAD

**Discussion by emmanouela88** [Megabit/Blazorise#4468] [Q&A] [answered]
I have a rich text editor and I am trying to validate the input a user is writing. Specifically to not be empty. I have created a TextEdit invisible component in order to get that message, but it seems that my validation on textedit to not be triggered. I have a fluentvalidations method, and is not a form.

Here is what I am doing:

<FieldLabel>My message:</FieldLabel>
                                <RichTextEdit @ref="rteMessage"
                                              ContentChanged="@MyCheckRTEMethod"
                                              PlaceHolder="Type your message here..."
                                              ReadOnly="@readOnly"
                                              SubmitOnEnter="false"
                                              Style="height:80px">
                                    <Editor>
                                       
                                    </Editor>
                                    <Toolbar>
                                        <RichTextEditToolbarGroup Float="Float.End">                                               
                                            <RichTextEditToolbarButton Action="RichTextEditAction.Bold" />
                                            <RichTextEditToolbarButton Action="RichTextEditAction.Italic" />
                                            <RichTextEditToolbarButton Action="RichTextEditAction.Strike" />
                                            <RichTextEditToolbarButton Action="RichTextEditAction.Image" />
                                            <RichTextEditToolbarButton Action="RichTextEditAction.Clean" />
                                        </RichTextEditToolbarGroup>
                                    </Toolbar>
                                </RichTextEdit>
                            </Field>
                        </Fields>
                        <Fields>
                            <Field>
                                <Validation AsyncValidator="@ValidateMessageInputAsync">
                                    <TextEdit @bind-Text="messageHiddenForValidation" @bind-Text:event="oninput" Visibility="Visibility.Invisible">
                                            <Feedback>
                                            <ValidationError>Please type a message.</ValidationError>
                                        </Feedback>
                                    </TextEdit>
                                </Validation>
                            </Field>
Has someone has a clue of what needs to be done or change in order to make it work? Thanks in advance

**Comment by stsrki** [comment, marked as answer]
The validation is not supported in RTE at the moment. Look at this answer for reference on how to workaround it https://github.com/Megabit/Blazorise/issues/1792#issuecomment-1369741666

**Reply by emmanouela88** [reply]
Oh cool, thanks. I couldn't find that answer to be honest when I was searching for the validations and rte. 

One more thing I would like to ask for the text edit. Even though I was able to show the input I was writing from the rte to the textedit component, how would I trigger the validation on the text edit? I tried with the TextChanged but nothing happened. Is there a way that I could trigger the input of a text edit from a different component or not?

**Reply by stsrki** [reply]
The validation should be updated automatically when the value is bound to TextEdit, eg. `Text="@someVariable"`. But, that only works in the upcoming v1.2, as we had to make some internal changes to make it work properly. The new version is expected to be released at the end of this month.

**Reply by emmanouela88** [reply]
Hmm I see, then I'll be waiting for your new release to see how it might work 😄 Thanks again for the help! Have a great day!
